src_py/apiServer/apiServer.py

```diff
@@ -413,11 +413,14 @@ def accuracy_matrix(self, expNum):
                 # print(f"worker {worker}, has {len(workerNeuronRes[worker][TRUE_LABLE_IND])} labels, with {len(workerNeuronRes[worker][TRUE_LABLE_IND][j])} samples")
                 # print(f"confusion {worker}:{j}, has is of {workerNeuronRes[worker][TRUE_LABLE_IND][j]}, {workerNeuronRes[worker][PRED_LABLE_IND][j]}")
                 confMatList[worker][j] = confusion_matrix(workerNeuronRes[worker][globe.TRUE_LABLE_IND][j], workerNeuronRes[worker][globe.PRED_LABLE_IND][j])
-                # print(confMatList[worker][j])
                 disp = ConfusionMatrixDisplay(confMatList[worker][j], display_labels=["X", labelNames[j]])
+                if confMatList[worker][j].shape == (0,0): # ! Worker is down
+                    disp = ConfusionMatrixDisplay(np.array([[0,0],[0,0]]), display_labels=["X", labelNames[j]])
+                    workerNeuronRes[worker][globe.TRUE_LABLE_IND][j] = [0,0]
+                    workerNeuronRes[worker][globe.PRED_LABLE_IND][j] = [1,1]
                 disp.plot(ax=axes[i, j], colorbar=False)
                 disp.ax_.set_title(f'{worker}, class #{j}\nAccuracy={round(accuracy_score(workerNeuronRes[worker][globe.TRUE_LABLE_IND][j], workerNeuronRes[worker][globe.PRED_LABLE_IND][j]), 3)}')
-                if i < len(workersList) - 1:
+                if i < len(workersList) - 1: # ? Ask David why this is needed
```

Comment:
@halfway258 Haran can you please clarify this line
@GuyPerets106  Please add a comment with description gave by Haran